ambry-api/src/main/java/com/github/ambry/store/MessageWriteSet.java

```diff
@@ -36,6 +38,12 @@ public interface MessageWriteSet {
   List<MessageInfo> getMessageSetInfo();
 
   /**
-   * Set life Version for {@link StoreKey}.
+   * Write the messages in this set to the given write channel asynchronously
+   * @param writeChannel The write interface to write the messages to
+   * @return a {@link CompletableFuture} that will eventually contain the size in bytes that was written to the
+   *         write interface or the {@link StoreException} if an error occurred
    */
-}
\ No newline at end of file
+  default CompletableFuture<Long> writeAsyncTo(Write writeChannel) {
```

Comment:
Will there be a case when both `writeTo` and `writeAsyncTo` should be present in the same class? If not I would suggest inherit MessageWriteSet into `SyncMessageWriteSet` and `AsyncMessageWriteSet`. The sync interface should have method `writeTo` and async should have method `writeAsyncTo`.

Reply:
Actually,  since asynchronous write method is only needed for cloud store, moved `writeAsyncTo` to `CloudMessageWriteSet` instead of having separate `AsyncMessageWriteSet`.